content/docs/use-cases/ci-cd-for-machine-learning.md

```diff
@@ -0,0 +1,69 @@
+# Continuous Integration and Deployment for Machine Learning
+
+MLOps and DataOps apply DevOps methodologies to machine learning (ML) and data
+management. The fields are rapidly evolving beyond a mere set of best practices,
+and now encompass the entire product lifecycle management. A core component is
+the automation of testing and deployment: CI/CD (continuous integration and
+continuous delivery).
+
+CI/CD is easy to set up with practically any Git repository. Some example
+scenarios are:
+
+- Agile development
+  - Automating quality assurance (QA) means teams can confidently deploy new
+    versions several times a day and even before the weekend without fear of
+    bugs/regressions
+- Easy testing
+  - When new data is pushed in a pull request (PR), validation tests
+    automatically run in the cloud (without each collaborator setting up
+    potentially inconsistent local testing environments)
+  - When a new model is pushed in a PR, reports (on performance against the
+    current deployed models) are automatically generated and posted as a PR
+    comment with graphs and tables
+- Continuous machine learning (CML): online training
+  - Schedule regular jobs pulling in data updates from public sources to refine
+    deployed models
+  - Provision hardware in the cloud and run a brute force hyperparameter search
+
+In ML, CI/CD can be used to automatically build, test and push data and models
```

Comment:
so, are we replacing this with a diagram?

Reply:
do you have an idea/improvements to the current diagram (based on the one I sent you recently)? We need to send it to someone to make it look good

Reply:
Which diagram did you send me?

I came up with this as a more concrete example:

![image](https://user-images.githubusercontent.com/10780059/116426935-77b7c080-a83b-11eb-90a1-c5f3ed248ce4.png)

Reply:
@DavidGOrtega please let me know if this is not quite accurate (or can be further simplified)

Reply:
@casperdcl lgtm! Maybe we are adding DVC here where it's not strictly necessary. I mean, it can be used but its not per se CML. 
Also it does not always have to deploy an instance... but a runner...so I would probably replace 3 4 5 for a remote or local runner just doing train.

Reply:
As in combine the `CI/CD Server` and `Spot instance` boxes into one? Sure, there are advantages to both but in terms of showcasing the config options it seems better to demo this more complex case. 

It's hopefully easy to tell from this diagram that an instance doesn't *have* to be deployed. If not, I'd prefer explaining so in text rather than simplifying the diagram (again, keeping target audience & showcasing features in mind). What do you think?

Reply:
Looks a bit low level like an architectural diagram. In other use cases we try to convey problem/solution visually with the figures. But I with @dberenbaum that we can leave diagrams for a further iteration once the case story (text) is stable.

Reply:
removed (preserved for later more in-depth tutorial etc)